compiler/src/main/java/com/squareup/anvil/compiler/codegen/PsiUtils.kt

```diff
@@ -402,6 +405,49 @@ internal fun ModuleDescriptor.findClassOrTypeAlias(
   return null
 }
 
+internal fun KtClassOrObject.typeVariableNames(
+  module: ModuleDescriptor
+): List<TypeVariableName> {
+  // Any type which is constrained in a `where` clause is also defined as a type parameter.
+  // It's also technically possible to have one constraint in the type parameter spot, like this:
+  // class MyClass<T : Any> where T : Set<*>, T : MutableCollection<*>
+  // Merge both groups of type parameters in order to get the full list of bounds.
+  val boundsByVariableName = mutableMapOf<String, MutableList<TypeName>>()
```

Comment:
Appen `.withDefault { mutableListOf() }`, then you can replace the `getOrPut`s below with `getValue(variableName)`.

Reply:
Unfortunately that doesn't work.  `withDefault { ... }` returns a default list, but doesn't actually do a `put` into to the map.  So after adding the `TypeName` to the default list, we'd still need to add a `put`.

Instead, I just used `associateTo` for the parameters.  Since any generic in the constraint list must also be at least named in the `parameters` list, we can still use `getValue` in the second `forEach`.

Reply:
Oh, I wasn't aware. Sounds good!